docs/config_reference.rst

```diff
@@ -1067,6 +1067,14 @@ General Configuration
    Ignore test name conflicts when loading tests.
 
 
+.. js:attribute:: .general[].ignore_exit_code
```

Comment:
I would make the feature a bit more general and call it `trap_job_errors` with default value `false`. When set, apart from what you are doing now, we should generate the job script with `trap_errors=True` here:

https://github.com/eth-cscs/reframe/blob/master/reframe/core/pipeline.py#L1317-L1320

This would cause the script to exit if any command in it fails.